app/src/main/java/im/rah/nightwear/NightWearTileService.kt

```diff
@@ -0,0 +1,183 @@
+package im.rah.nightwear
+
+import android.util.Log
+import androidx.core.content.ContextCompat
+import androidx.preference.PreferenceManager
+import androidx.wear.tiles.ActionBuilders
+import androidx.wear.tiles.ColorBuilders.argb
+import androidx.wear.tiles.DeviceParametersBuilders.DeviceParameters
+import androidx.wear.tiles.DimensionBuilders.*
+import androidx.wear.tiles.LayoutElementBuilders.*
+import androidx.wear.tiles.ModifiersBuilders
+import androidx.wear.tiles.RequestBuilders.ResourcesRequest
+import androidx.wear.tiles.RequestBuilders.TileRequest
+import androidx.wear.tiles.ResourceBuilders.Resources
+import androidx.wear.tiles.TileBuilders.Tile
+import androidx.wear.tiles.TileService
+import androidx.wear.tiles.TimelineBuilders.Timeline
+import androidx.wear.tiles.TimelineBuilders.TimelineEntry
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.guava.future
+
+private const val RESOURCES_VERSION = "1"
+
+
+private const val BG_UNIT_MG = "mg/dL"
+private const val BG_UNIT_MMOL = "mmol/l"
+
+private const val BG_RANGE_LOW = 60;
+private const val BG_RANGE_HIGH = 210;
+
+private const val ARC_TOTAL_DEGREES = 360f
+private const val TILE_REFRESH_INTERVAL_MINUTES = 1
+private const val ID_CLICK_REFRESH_BG = "click_refresh_bg"
+private val PROGRESS_BAR_THICKNESS = dp(6f)
+
+
+/**
+ * Creates a tile, showing current blood glucose value and the trend arrow.
+ */
+class NightWearTileService : TileService() {
+    // For coroutines, use a custom scope we can cancel when the service is destroyed
+    private val serviceScope = CoroutineScope(Dispatchers.IO)
+
+    override fun onTileRequest(requestParams: TileRequest) = serviceScope.future {
+
+        if (requestParams.state!!.lastClickableId == ID_CLICK_REFRESH_BG) {
+            Log.d(tag,"force refresh bg value")
+            bloodGlucoseService.refresh(true)
```

Comment:
This is neat - I've contemplated similar in the past, it's comforting to be able to say "get me the latest thing from nightscout right now".